Deployment tests on mac/py3.11/openeye are failing

https://github.com/openforcefield/openff-toolkit/actions/workflows/conda.yml

Several errors, which seem to have the same root cause. Specifically, hundreds of parameter assignment tests:
```
FAILED openff/toolkit/_tests/test_forcefield.py::TestForceFieldParameterAssignment::test_molecule_energy_gb_no_sa[OBC2-False] - ModuleNotFoundError: No module named 'numpy.compat'
```
and one more vanilla import
```
FAILED openff/toolkit/_tests/test_forcefield.py::TestForceFieldParameterAssignment::test_multi_alkethoh_parameters_assignment - ModuleNotFoundError: No module named 'numpy.compat'
```

--- a/devtools/conda-envs/conda.yaml
+++ b/devtools/conda-envs/conda.yaml
@@ -8,3 +8,5 @@ dependencies:
   - pytest=7.4
   - pytest-rerunfailures
   - nbval
+  # Remove once https://github.com/ParmEd/ParmEd/pull/1387 gets into a release
+  - numpy <2.3

--- a/devtools/conda-envs/conda_oe.yaml
+++ b/devtools/conda-envs/conda_oe.yaml
@@ -10,3 +10,5 @@ dependencies:
   - pytest=7.4
   - pytest-rerunfailures
   - nbval
+  # Remove once https://github.com/ParmEd/ParmEd/pull/1387 gets into a release
+  - numpy <2.3

